core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp

```diff
@@ -292,6 +385,11 @@ void VirtualSwapchain::Destroy(const VkAllocationCallbacks *pAllocator) {
   }
 
   functions_->vkDestroyCommandPool(device_, command_pool_, pAllocator);
+
+  if (callback_user_data_ != nullptr) {
```

Comment:
Same as above, if this comes from the user, we will be deleting a user object. That is very bad.